direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorageTest.java

```diff
@@ -631,34 +631,93 @@ public boolean onError(Throwable error) {
     assertEquals(3, observeHandle.getCurrentOffsets().size());
   }
 
-  @Test(expected = UnsupportedOperationException.class)
+  @Test
+  public void testObserveSinglePartitionOutOfMultiplePartitions() throws InterruptedException {
```

Comment:
This test looks it would pass even when all the data would be in single partition. Can we add a check for the keys that we process?

Reply:
fixed